### PR TITLE
Dependency upgrades

### PR DESCRIPTION
**Pull request by KieranWardle**
# Motivation and Context
The python version and most dependencies were very out of date

# What has changed
Upgraded all dependencies and the python version
Removed old directories

# How to test?
Tested through acceptance tests

# Links
https://trello.com/c/y6xA2Djt

# Screenshots (if appropriate):

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.6
+FROM python:3.10.4
```
Comment (AdamHawtin) [inline Dockerfile:1]: Could we swap to `python:3.10.4-slim` here? Should work as a straight swap, we've already checked it's fine on the slim image.

File: Pipfile [hunk 2]
```diff
@@ -4,37 +4,38 @@ verify_ssl = true
...
+aiohttp-session = {version = "~=2.11.0", extras = ["aioredis"]}
 aiohttp-utils = "~=3.1.1"
-aiohttp = "~=3.7.4"
-aioredis = "~=1.3.1"
-babel = "==2.9.1"
-cchardet = "~=2.1.6"
+aiohttp = "~=3.8.1"
+aioredis = "~=2.0.1"
+babel = "==2.10.1"
+cchardet = "~=2.1.7"
```
Comment (AdamHawtin) [inline Pipfile:13]: I think `cchardet` is unused, tested locally and unit/cucumber tests pass without it

File: Pipfile [hunk 3]
```diff
@@ -4,37 +4,38 @@ verify_ssl = true
...
-invoke = "~=1.4.1"
-iso8601 = "~=0.1.12"
-python-json-logger = "~=0.1.11"
-structlog = "~=20.1.0"
-tenacity = "~=6.2.0"
-jinja2 = "~=2.11.2"
-importlib-metadata = "~=4.8.1"
-urllib3 = "~=1.26.5"
+invoke = "~=1.7.0"
+iso8601 = "~=1.0.2"
```
Comment (AdamHawtin) [inline Pipfile:16]: I think `iso8601` is unused, tested locally and unit/cucumber tests pass without it

**Review by AdamHawtin — CHANGES_REQUESTED** [review]

File: app/session.py [hunk 4]
```diff
@@ -38,27 +39,24 @@ def aiohttp_session_pr_331_rollback(self, identity, *, data, new,
 
 def setup(app_config):
     # Monkey patch aiohttp_session.py Session.__init__ method to remove PR 331 as above
-    Session.__init__ = aiohttp_session_pr_331_rollback
+    # Session.__init__ = aiohttp_session_pr_331_rollback
```
Comment (AdamHawtin) [inline app/session.py:42]: Looks like we can delete this entirely now, including the function above that it was patching in. I checked the issue linked in the comment, and it's been resolved/closed for some time.

**Review by AdamHawtin — CHANGES_REQUESTED** [review]
Awesome work getting it all working and up to date, built and tested locally and all good 👍 

Just one bit of commented out/now dead code I think should be deleted.

**Review by AdamHawtin — APPROVED** [review]

**Review by hbrace — APPROVED** [review]

**Pull request merged**